dialogue-annotations/src/main/java/com/palantir/dialogue/annotations/HttpEntityBodyRequestBodyAdapter.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.annotations;
+
+import com.palantir.dialogue.RequestBody;
+import com.palantir.logsafe.UnsafeArg;
+import java.io.IOException;
+import java.io.OutputStream;
+import org.apache.hc.core5.http.HttpEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Adapts {@link HttpEntity} to a {@link RequestBody}.
+ */
+abstract class HttpEntityBodyRequestBodyAdapter implements RequestBody {
```

Comment:
Thought's on separate subproject for these? I guess Json is already here...

Reply:
I don't have a strong opinion either way -- I suppose the question is whether we'd expect these to ever be used without the dialogue annotation processor? I'd hope not, so this is a reasonable enough place for them to live.

Reply:
I was going to call it dialogue-annotation-processor-extras or something, just to lighten the dependency. But then I realized that people already depend on apache, so it's a moot point. 

>  I'd hope not, so this is a reasonable enough place for them to live.

No, we do not want anyone to use this outside of annotation processor.